integration_tests/suite/test_http_callback.py

```diff
@@ -548,3 +550,132 @@ def test_given_http_subscription_with_wazo_uuid_when_bus_events_then_only_callba
             count=1,
             exact=True,
         ), tries=10, interval=0.5)
+
+    @subscription(TEST_SUBSCRIPTION)
+    def test_given_one_http_subscription_check_pagination(self, subscription):
+        self.third_party.reset()
+        self.third_party.mock_simple_response(
+            path='/test',
+            responseBody="temporary bugged service",
+            statusCode=503,
+        )
+        self.third_party.mock_simple_response(
+            path='/test',
+            responseBody="temporary bugged service",
+            statusCode=503,
+        )
+        self.third_party.mock_simple_response(
+            path='/test',
+            responseBody="working service",
+            statusCode=200,
+        )
+        self.bus.publish(trigger_event(),
+                         routing_key=SOME_ROUTING_KEY,
+                         headers={'name': TRIGGER_EVENT_NAME})
+
+        until.assert_(self.make_third_party_verify_callback(
+            request={'method': 'GET', 'path': '/test'},
+            count=3,
+            exact=True
+        ), tries=20, interval=0.5)
+
+        self.third_party.reset()
+        self.third_party.mock_simple_response(
+            path='/test',
+            responseBody="temporary bugged service",
+            statusCode=503,
+        )
+        self.third_party.mock_simple_response(
+            path='/test',
+            responseBody="working service",
+            statusCode=200,
+        )
+
+        self.bus.publish(trigger_event(),
+                         routing_key=SOME_ROUTING_KEY,
+                         headers={'name': TRIGGER_EVENT_NAME})
+
+        until.assert_(self.make_third_party_verify_callback(
+            request={'method': 'GET', 'path': '/test'},
+            count=2,
+            exact=True
+        ), tries=20, interval=0.5)
+
+        webhookd = self.make_webhookd(MASTER_TOKEN)
+
+        # Default order
+        logs = webhookd.subscriptions.get_logs(subscription["uuid"])
+        self.assertEqual(5, logs['total'])
+        self.assertEqual(
```

Comment:
Our standard is to use hamcrest everywhere for the readability. The only place where there are no hamcrest is in legacy code. Hamcrest avoid non-trivial assert like  `list(map(operator.itemgetter...`

Reply:
I changed them. But honestly I mitigated about hamcrest, I agree it improves the readability of the test file, but the assert reporting is horrible, I have always to add code to understand what entry is missing or wrong...

Reply:
I agreed that some time, hamcrest is not really clear about the error. There are probably better solution (from pytest?).
My point is that if a test is hard to read, it is not maintained. And I consider the test maintenance more common then a failure